File: models/src/main/java/ar/edu/itba/paw/models/Patient.java
package ar.edu.itba.paw.models;

import javax.persistence.*;

@Entity
@Table(
        name = "patient",
        indexes = {
                @Index(columnList = "patient_id", name = "patient_patient_id_uindex", unique = true),
        }
)
public class Patient extends GenericModel<Integer> {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "patient_pk")
    @SequenceGenerator(sequenceName = "patient_pk", name = "patient_pk", allocationSize = 1)
    @Column(name = "patient_id")
    private Integer id;
    @JoinColumn(name = "user_id")
    @ManyToOne(fetch = FetchType.EAGER)
    private User user;
    @JoinColumn(name = "office_id")
    @ManyToOne(fetch = FetchType.EAGER)
    private Office office;

    @Override
    public Integer getId() {
        return this.id;
    }

    @Override
    public void setId(Integer id) {
        this.id = id;
    }

    public User getUser() {
        return this.user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Office getOffice() {
        return this.office;
    }

    public void setOffice(Office office) {
        this.office = office;
    }

    @Override
    protected boolean isSameInstance(Object o) {
        return o instanceof Patient;
    }
}

File: vueapp/src/logic/models/Patient.ts
import {GenericEntity} from '~/logic/models/utils/GenericEntity';

export class Patient extends GenericEntity<Patient> {
    private _id: number;
    private _userId: number;
    private _officeId: number;

    public get id(): number {
        return this._id;
    }

    public set id(value: number) {
        this._id = value;
    }

    public get userId(): number {
        return this._userId;
    }

    public set userId(value: number) {
        this._userId = value;
    }

    public get officeId(): number {
        return this._officeId;
    }

    public set officeId(value: number) {
        this._officeId = value;
    }
}
